dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/handler/DataHandler.java

```diff
@@ -309,10 +303,17 @@ private IndicatorValue getIndicatorValue( List<Period> filterPeriods, Map<String
             ? permutationOrgUnitTargetMap.get( ou )
             : null;
 
-        return expressionService.getIndicatorValueObject( indicator, periods,
+        return expressionService.getIndicatorValueObject( removeCategoryOptionPrefix( indicator ), periods,
             convertToDimItemValueMap( valueMap ), constantMap, orgUnitCountMap );
     }
 
+    private Indicator removeCategoryOptionPrefix( Indicator indicator )
+    {
+        indicator.setNumerator( indicator.getNumerator().replace( "co:", "" ) );
+        indicator.setDenominator( indicator.getDenominator().replace( "co:", "" ) );
+        return indicator;
+    }
+
```

Comment:
This is dangerous. The character sequence "co:" could appear elsewhere in an expression such as inside a string that is being compared with a code. We might also use it in other places in the future.

Could this be avoided by adding the "co:" prefix in `DataElementOperand.getDimensionItem()`, as suggested above?

Reply:
changed